### PR TITLE
Recreate sts after resize all PVCs

### LINKED ISSUE #4547: No need to resize PVC for new Pods after scaling up PVC
## Feature Request

**Is your feature request related to a problem? Please describe:**

Currently, TiDB Operator supports resizing PVC for the components, however, if I increase the PVC of TiKV from 100G to 500G, after that, I scale out 1 more TiKV, the new TiKV will use 100G PVC first, and then TiDB Operator will increase its PVC to 500G.
So it's better that the new TiKV use 500G PVC when it's created.
**Describe the feature you'd like:**

**Describe alternatives you've considered:**

**Teachability, Documentation, Adoption, Migration Strategy:**

### PR DESCRIPTION
**Pull request by KanShiori**
### What problem does this PR solve?

Close #4547
### Code changes

- [x] Has Go code change
- [ ] Has CI related scripts change

### Tests

- [x] Unit test 
- [ ] E2E test 
- [x] Manual test 

For every component:
1. Scale up the storage size from 200Gi to 300Gi.
2. `Resizing` condition is set to true.
3. Check that PVCs are resizing pod by pod.
4. Check that STS is recreated after all PVCs are resized.
5. `Resizing` condition is set to false.
6. Scale out replication, check the size of new PVC is expected.

For storage class not support expansion:
1. Scale up the storage size for TiKV from 200Gi to 300Gi.
2. `Resizing` condition is set to true.
4. Check STS is recreated with 200Gi size, but PVCs are not changed.
7. `Resizing` condition is set to false.
8. Scale out TiKV, check the size of  new PVC is 300Gi.

- [ ] No code 

### Side effects

- [ ] Breaking backward compatibility 
- [ ] Other side effects: 

### Related changes

- [ ] Need to cherry-pick to the release branch 
- [ ] Need to update the documentation 

### Release Notes

Please refer to [Release Notes Language Style Guide](https://github.com/pingcap/tidb-operator/blob/master/docs/release-note-guide.md) before writing the release note.

```release-note
Recreate statefulset after resize all PVCs for a components
```

### CODE REVIEW

File: pkg/manager/member/pvc_resizer.go [hunk 1]
```diff
@@ -391,29 +392,60 @@ func (p *pvcResizer) updateVolumeStatus(ctx *componentVolumeContext) {
 
 // resizeVolumes resize PVCs by comparing `desiredVolumeQuantity` and `actualVolumeQuantity` in context.
 func (p *pvcResizer) resizeVolumes(ctx *componentVolumeContext) error {
+	ns := ctx.cluster.GetNamespace()
+	name := ctx.cluster.GetName()
+
+	stsName := controller.MemberName(name, ctx.status.GetMemberType())
+	sts, err := p.deps.StatefulSetLister.StatefulSets(ns).Get(stsName)
+	if err != nil {
+		return fmt.Errorf("failed to get sts %s: %s", stsName, err)
```
Comment (csuzhangxc) [inline pkg/manager/member/pvc_resizer.go:401]: include `ns` in the error message?

**Review by csuzhangxc — APPROVED** [review]

**Review by csuzhangxc — APPROVED** [review]

**Review by DanielZhangQD — APPROVED** [review]
LGTM

**Pull request merged**